scripts/tests/test_docker_push_with_retry.py

```
@@ -0,0 +1,282 @@
+          fake_bin=fake_bin,
+          image_ref="ghcr.io/example/wave:test",
+          extra_env={"DOCKER_PUSH_MAX_ATTEMPTS": "3"},
+      )
+
+      self.assertNotEqual(0, result.returncode)
+      self.assertEqual("3", attempt_file.read_text(encoding="utf-8"))
+      self.assertIn("unknown blob", f"{result.stdout}\n{result.stderr}")
+      self.assertIn("failed after 3 attempts", f"{result.stdout}\n{result.stderr}")
+
+  def test_build_mock_command_env_prepends_fake_bin_and_resolved_bash_dir(self):
+    fake_bin = Path("/tmp/fake-bin")
```

Ruff (0.15.9)

[error] Probable insecure usage of temporary file or directory: "/tmp/fake-bin"

(S108)